WireDomain/Tests/WireDomainTests/Synchronization/PullAllConversationsSyncTests.swift

```diff
@@ -0,0 +1,206 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireAPISupport
+import XCTest
+@testable import WireAPI
+@testable import WireDomain
+@testable import WireDomainSupport
+
+final class PullAllConversationsSyncTests: XCTestCase {
+
+    private var sut: PullAllConversationsSync!
+    private var api: MockConversationsAPI!
+    private var store: MockConversationLocalStoreProtocol!
+
+    override func setUp() async throws {
+        api = MockConversationsAPI()
+        store = MockConversationLocalStoreProtocol()
+        sut = PullAllConversationsSync(
+            localDomain: Scaffolding.localDomain,
+            isFederationEnabled: Scaffolding.isFederationEnabled,
+            isMLSEnabled: Scaffolding.isMLSEnabled,
+            api: api,
+            store: store
+        )
+    }
+
+    override func tearDown() async throws {
+        api = nil
+        store = nil
+        sut = nil
+    }
+
+    func testPull() async throws {
+        // Mock
+        api.getConversationIdentifiers_MockValue = .init(fetchPage: { _ in
+            .init(
+                element: Scaffolding.conversationIDs,
+                hasMore: false,
+                nextStart: .init()
+            )
+        })
+
+        api.getConversationsFor_MockValue = .init(
+            found: [Scaffolding.remoteConversation1],
+            notFound: [Scaffolding.conversationID2],
+            failed: [Scaffolding.conversationID3]
+        )
+
+        store.storeConversationTimestampIsFederationEnabledIsMLSEnabled_MockMethod = { _, _, _, _ in }
+        store.storeConversationNeedsBackendUpdateConversationIDConversationDomain_MockMethod = { _, _, _ in }
+        store.storeFailedConversationConversationIDConversationDomain_MockMethod = { _, _ in }
+
+        // When
+        try await sut.pull()
+
+        // Then
+        XCTAssertEqual(api.getConversationsFor_Invocations.count, 1)
+        XCTAssertEqual(
+            store.storeConversationTimestampIsFederationEnabledIsMLSEnabled_Invocations.count,
+            1
+        )
+
+        // Then
```

Comment:
`// Then` is 2 times

Reply:
removed 6d6af8b3a13ed04e4a3a7ee0829126c90ce9801f